### PR TITLE
feat(steam-user): update to v4.23

### PR DESCRIPTION
**Pull request by joshuajeschek**
* I updated the types to the recently released version 4.21. DoctorMcKay outlines the changes [here](https://github.com/DoctorMcKay/node-steam-user/releases/tag/v4.21.0)
* I updated the types to the shortly-thereafter-released version 4.22. Changes are outlined here:
    * [release v4.21.0](https://github.com/DoctorMcKay/node-steam-user/releases/tag/v4.22.0)
    * [release v4.22.1](https://github.com/DoctorMcKay/node-steam-user/releases/tag/v4.22.1)
    * https://github.com/DoctorMcKay/node-steam-user/pull/378
    * [release v4.23.0](https://github.com/DoctorMcKay/node-steam-user/releases/tag/v4.23.0)
* Furthermore I added the event `chatroomgrouproomschange` to the `SteamChatRoomClient`, since I seemed to have missed that previously.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
    - https://github.com/DoctorMcKay/node-steam-user/releases/tag/v4.21.0
    - https://github.com/DoctorMcKay/node-steam-user/wiki/SteamChatRoomClient#chatroomgrouproomschange
    - see complete list at the beginning of this message
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## steam-user ([<kbd>unpkg</kbd>](https://unpkg.com/browse/steam-user@latest/))
was missing the following properties:
1. CurrencyData
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 44a2f7e3b8c3ef434599b94b16b73c56a9237250
</p>

**Comment by Revadike** [comment]
The package has just been updated, can this PR be updated accordingly. Thanks in advance!

**Comment by joshuajeschek** [comment]
Yes, the changes are underway! I was just unsure about how the filter function should look, but now I know it is like an Array filter :)

**Comment by joshuajeschek** [comment]
@Revadike can you confirm that this is the way the filter function works? Have I missed something? Wasn't quite sure if there was a way to import the definition from `Array.prototyp.filter` and add the needed type.

If you want to, you could also become a co-owner of the types here, since it is always difficult to get these PRs through alone, and I know you are also active in steam-user :) If not that's also fine :)

**Comment by Revadike** [comment]
The thing is, `filter` can be a function of type `Array.propotype.filter`, or an object with 3 boolean key-value pairs. I'm just a contributor to `steam-user`. I don't often use typescript. I only came across this, because we talked about it on [discord](https://discord.gg/js28PxBwHk).

**Comment by joshuajeschek** [comment]
Thanks for your clarification! I think the types I wrote should be fine then - I was just confused by the wording "of type `Array.propotype.filter`", because in reality it is just the part:
```ts
(element: T, index: number, array: T[]) => boolean
```
right?
I'll also check out the discord, as I seem to be the one to talk to about typescript problems :)

File: types/steam-user/index.d.ts [hunk 1]
```diff
@@ -859,7 +867,33 @@ interface Events {
...
+type RegionCode = 0x00 | 0x01 | 0x02 | 0x03 | 0x04 | 0x05 | 0x06 | 0x07 | 0xFF; // https://developer.valvesoftware.com/wiki/Master_Server_Query_Protocol#Region_codes
+type OwnsFilterFunction = (element: Proto_CMsgClientLicenseList_License, index: number, array: Proto_CMsgClientLicenseList_License[]) => boolean;
+interface Proto_CMsgClientLicenseList_License {
+    package_id: number;
+    time_created: number;
+    time_next_process: number;
+    minute_limit: number;
+    minutes_used: number;
+    payment_method: SteamUser.EPaymentMethod;
+    flags: number;
```
Comment (DoctorMcKay) [inline types/steam-user/index.d.ts:879]: FYI I just pushed a jsdoc change to properly refer to `flags` and `license_type` with their enum types: https://github.com/DoctorMcKay/node-steam-user/commit/9312326c34fed69a2ea6c4102ed25e3073f0516d

I'm not going to bother publishing such a minor doc change to npm, but you might want to go ahead and change these here.

Comment (joshuajeschek) [reply types/steam-user/index.d.ts:879]: Thanks for letting me know!

**Comment by Revadike** [comment]
Those were only documentation changes. ^^

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Review by ruj — CHANGES_REQUESTED** [review]
Thanks for the quick fix, I would like to add that this property **dataDirectory** can also be given a **null** value so that the data is not saved in a directory.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by joshuajeschek** [comment]
ready to merge

thanks!

**Pull request merged**